src/chatdDb.h

```diff
@@ -364,6 +364,54 @@ class ChatdSqliteDb: public chatd::DbInterface
         msg.assign(buf, stmt.intCol(0), stmt.uint64Col(3), stmt.intCol(1), stmt.uint64Col(4));
     }
 
+    virtual void setOpenChat()
+    {
+        mDb.query(
+            "insert or replace into chat_vars(chatid, name, value) "
+            "values(?, 'open_chat', true)", mChat.chatId());
+        assertAffectedRowCount(1);
+    }
+
+    //Remove open_chat and preview_mode records related to the chatId
+    virtual void removeOpenChat()
+    {
+        mDb.query("delete from chat_vars where chatid = ?", mChat.chatId());
+        assertAffectedRowCount(1);
+    }
+
+    virtual bool openChat()
+    {
+        SqliteStmt stmt(mDb, "select value from history where chatid = ? and name=open_chat");
+        stmt << mChat.chatId();
+        return (stmt.step()) ? stmt.int64Col(0) : false;
```

Comment:
Check datatypes for each column in DB schema. Value is not an `int64` but a text string (despite it has the value `1`)
https://github.com/meganz/MEGAchat/blob/develop/src/dbSchema.sql#L26